dm/syncer/validator_checkpoint.go

```diff
@@ -335,34 +305,106 @@ func (c *validatorPersistHelper) persist(tctx *tcontext.Context, loc binlog.Loca
 			args = append(args, []interface{}{
 				c.cfg.SourceID, sourceTable.Schema, sourceTable.Table, r.srcJob.Key,
 				targetTable.Schema, targetTable.Table,
-				srcDataStr, dstDataStr, r.tp, pb.ValidateErrorState_NewErr,
+				string(srcDataStr), string(dstDataStr), r.tp, pb.ValidateErrorState_NewErr,
```

Comment:
so `srcDataStr` is not string, maybe  rename this to `srcData` and `dstData`

Reply:
renamed https://github.com/pingcap/tiflow/pull/5397/commits/2d6ee662e21f19f7a9b33251db86483d7b2d8219